llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -210,36 +215,100 @@ void InterBlockScheduling::leaveFunction() {
 }
 
 void InterBlockScheduling::enterBlock(MachineBasicBlock *BB) {
-  CurrentBlock = &getBlockState(BB);
-  CurrentBlock->resetRegion();
+  CurrentBlockState = &getBlockState(BB);
+  CurrentBlockState->resetRegion();
   DEBUG_BLOCKS(dbgs() << "  >> enterBlock " << BB->getNumber() << " "
-                      << CurrentBlock->kindAsString() << " FixPointIter="
-                      << CurrentBlock->FixPoint.NumIters << "\n");
-}
+                      << CurrentBlockState->kindAsString() << " FixPointIter="
+                      << CurrentBlockState->FixPoint.NumIters
+                      << " II=" << CurrentBlockState->FixPoint.II << "\n");
+}
+namespace {
+class PipelineExtractor : public PipelineScheduleVisitor {
```

Comment:
Nit: Maybe add a high-level description?

I think we use this as a helper to insert instructions (or their clones) at the right places (prologue, steady state, epilogue)